.github/workflows/build_and_test_dbt.yaml

```diff
@@ -31,14 +41,17 @@ jobs:
           path: ${{ env.PROJECT_DIR }}/${{ env.STATE_DIR }}
           key: ${{ env.CACHE_KEY }}
 
-      - if: steps.cache.outputs.cache-hit == 'true'
-        name: Set command args to build/test modified resources
-        run: echo "MODIFIED_RESOURCES_ONLY=true" >> "$GITHUB_ENV"
-        shell: bash
+        name: Set command args for building/testing resources
+        run: |
+          if [[ ${{ steps.cache.outputs.cache-hit == 'true' }} ]]; then
+            echo "MODIFIED_RESOURCES_ONLY=true" >> "$GITHUB_ENV"
+          else
+            echo "MODIFIED_RESOURCES_ONLY=false" >> "$GITHUB_ENV"
+          fi
 
-      - if: steps.cache.outputs.cache-hit != 'true'
-        name: Set command args to build/test all resources
-        run: echo "MODIFIED_RESOURCES_ONLY=false" >> "$GITHUB_ENV"
+          if [[ ${{ github.event_name == 'workflow_dispatch' }} ]]; then
+              echo "MANUALLY_DISPATCHED=true" >> "$GITHUB_ENV"
+          fi
```

Comment:
I condensed these into a single step just for the sake of brevity, but if there's a reason to have them separate let me know!